### PR TITLE
[16.0][MIG] sale_margin_delivered

### PR DESCRIPTION
**Pull request by aliciagaarzo**
Migration of **sale_margin_delivered** module to version 16

### CODE REVIEW

**Comment by aliciagaarzo** [comment]
Waiting for module migration **sale_margin_security** https://github.com/OCA/margin-analysis/pull/173

File: sale_margin_delivered/views/sale_margin_delivered_view.xml
Comment (HugoCordobaLeal) [inline sale_margin_delivered/views/sale_margin_delivered_view.xml]: #173 is already merged, can you add this again?

**Review by dreispt — APPROVED** [review]

**Review by fcvalgar — APPROVED** [review]
Functional review.
LGTM, thank you @aliciagaarzo 👍

image


image

**Review by fcvalgar — CHANGES_REQUESTED** [review]
@aliciagaarzo I think the fields are displayed even if you do not have Show Sale Margin permissions.

Captura desde 2023-08-09 16-33-21

@moduon MT-3399

**Comment by rafaelbn** [comment]
Hello @sergio-teruel @chienandalu @CarlosRoca13 , do you want to review before merging as maintainers and contributors? Thank you!

File: sale_margin_delivered/views/sale_margin_delivered_view.xml
Comment (chienandalu) [inline sale_margin_delivered/views/sale_margin_delivered_view.xml]: You have to add the groups to fields themselves. Not just removing them from the view.

File: sale_margin_delivered/reports/sale_report.py
Comment (chienandalu) [inline sale_margin_delivered/reports/sale_report.py:17]: This one's a little picky, but more in line with the previous nomenclature
```suggestion
        fields = super()._select_additional_fields()
        fields[
            "margin_delivered"
        ] = "SUM(l.margin_delivered / COALESCE(s.currency_rate, 1.0))"
        return fields
```

File: sale_margin_delivered/reports/sale_report.py
Comment (chienandalu) [inline sale_margin_delivered/reports/sale_report.py:16]: Follow the guideline for currency conversion in reports: https://github.com/odoo/odoo/blame/116322b705c96a6f23167d09c8f42b569a9e6f22/addons/sale_margin/report/sale_report.py#L15-L16

**Review by chienandalu — CHANGES_REQUESTED** [review]
Hi @aliciagaarzo thanks for you work.

Some comments bellow :)

cc @Tecnativa TT44880

**Comment by sergio-teruel** [comment]
Ping @aliciagaarzo Can you anttend pending comments. Thanks

**Comment by pilarvargas-tecnativa** [comment]
Superseded by: https://github.com/OCA/margin-analysis/pull/183

**Pull request closed without merging**
